### PR TITLE
chore: Use queryEditorId in SqlEditor child components

### PR DESCRIPTION
**Pull request by EugeneTorap**
### SUMMARY

Fix potential re-rendering in SqlEditor child components.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/SqlLab/components/SaveQuery/index.tsx
Comment (justinpark) [inline superset-frontend/src/SqlLab/components/SaveQuery/index.tsx:82]: nit:
```suggestion
const SaveQuery : React.FC<SaveQueryProps> = ({
  queryEditorId,
  onSave = () => {},
  onUpdate,
  saveQueryWarning = null,
  database,
  columns,
}) => {

```

File: superset-frontend/src/SqlLab/components/SqlEditorLeftBar/index.tsx
Comment (justinpark) [inline superset-frontend/src/SqlLab/components/SqlEditorLeftBar/index.tsx:286]: this should be 
```suggestion
         onClick={() => dispatch(resetState())}
```

File: superset-frontend/src/SqlLab/components/SqlEditorLeftBar/index.tsx
Comment (justinpark) [inline superset-frontend/src/SqlLab/components/SqlEditorLeftBar/index.tsx:111]: `queryEditorRef` won't be no longer needed. Please use queryEditor (from line 107) instead.

Please remove the following block around line 135.

```
  useEffect(() => {
    queryEditorRef.current = queryEditor;
  }, [queryEditor, database]);```

File: superset-frontend/src/SqlLab/components/SqlEditorLeftBar/index.tsx
Comment (justinpark) [inline superset-frontend/src/SqlLab/components/SqlEditorLeftBar/index.tsx:249]: this causes infinite re-rendering.
Please use `useCallback`

File: superset-frontend/src/SqlLab/components/SqlEditorLeftBar/index.tsx
Comment (justinpark) [reply superset-frontend/src/SqlLab/components/SqlEditorLeftBar/index.tsx:286]: and please `useCallback`

**Comment by EugeneTorap** [comment]
@jinghua-qa @andrey-zayats Can you test the SQL Lab?

**Comment by EugeneTorap** [comment]
@michael-s-molina Can also you review the PR?

File: superset-frontend/src/SqlLab/components/SqlEditorLeftBar/SqlEditorLeftBar.test.jsx
Comment (michael-s-molina) [inline superset-frontend/src/SqlLab/components/SqlEditorLeftBar/SqlEditorLeftBar.test.jsx:63]:
```suggestion
  const renderAndWait = (props, store) => waitFor(() => setup(props, store));
```

File: superset-frontend/src/SqlLab/components/SqlEditorLeftBar/SqlEditorLeftBar.test.jsx
Comment (michael-s-molina) [inline superset-frontend/src/SqlLab/components/SqlEditorLeftBar/SqlEditorLeftBar.test.jsx:62]: There are some unnecessary uses of the await keyword in the tests. I created a patch file with some changes. To use it you need to uncompress the file and `git apply tests.patch`

[tests.patch.zip](https://github.com/apache/superset/files/9737632/tests.patch.zip)

**Comment by EugeneTorap** [comment]
@lyndsiWilliams Can you review and merge it?

File: superset-frontend/src/SqlLab/components/SqlEditorLeftBar/index.tsx
Comment (michael-s-molina) [inline superset-frontend/src/SqlLab/components/SqlEditorLeftBar/index.tsx:221]: Can you add the correct dependencies to `useCallback` functions?

Comment (michael-s-molina) [reply superset-frontend/src/SqlLab/components/SqlEditorLeftBar/index.tsx:221]: You'll see which ones I'm referring to by checking the warnings 😉

**Review by michael-s-molina — APPROVED** [review]
Thanks for the PR @EugeneTorap.

Code LGTM. We need @jinghua-qa's approval before merging it.

**Comment by EugeneTorap** [comment]
@jinghua-qa @andrey-zayats Can you test the SQL Lab?

**Comment by jinghua-qa** [comment]
test sqllab regression, LGTM!

**Review by jinghua-qa — APPROVED** [review]

**Pull request merged**
